tldw_Server_API/app/core/MCP_unified/modules/implementations/filesystem_module.py

```
@@ -0,0 +1,252 @@
+        workspace_root, target = await self._resolve_target_path(
+            str(arguments["path"]),
+            context=context,
+        )
+        if not target.exists():
+            raise FileNotFoundError(f"Path not found: {target}")
+        if not target.is_file():
+            raise ValueError("path must resolve to a file")
+
+        payload = target.read_bytes()
+        if _is_binary_payload(payload):
+            raise ValueError("binary file cannot be read with fs.read_text")
```

Ruff (0.15.7)

[warning] Avoid specifying long messages outside the exception class

(TRY003)